kore/test/Test/Kore/Equation/Sentence.hs

```diff
@@ -32,12 +32,12 @@ test_fromSentenceAxiom =
         assertions original equation
     , testCase "⌈f(x))⌉ → f(x) = g(x) ∧ ⌈h(x)⌉" $ do
         let requires = mkCeil sortR (Mock.f Mock.a)
-            ensures = mkCeil sortR (Mock.h Mock.b)
+            ensures = mkCeil Mock.testSort (Mock.h Mock.b)
             left = Mock.f (mkElemVar Mock.x)
             right = Mock.g (mkElemVar Mock.x)
             original =
                 mkImplies requires $
-                    mkAnd (mkEquals sortR left right) ensures
+                    mkEquals sortR left (mkAnd right ensures)
```

Comment:
Related to the comment above: we shouldn't change the old tests, just add new tests for the new encoding.

Reply:
I tried initially to just add new tests but eventually realized that they cannot coexist because of the `toTermLike` function.
We cannot support unparsing of both formats at the same time as we don't know for a particular equation whether it came from the old format or the new one.
I suppose we could add a flag inside the Equation data structure telling us which format was used for it's parsing for I'm not sure if this is the way to go...
What do you think?

Reply:
We don't need to support both at the same time, we should only support one. Right now, the frontend emits the old encoding, that's why I suggested [this](https://github.com/kframework/kore/pull/2688#discussion_r658954577). When the frontend emits the new encoding, we should just switch to using the new `toTermLike` and remove the old tests.